packages/opentelemetry-api/src/trace/Sampler.ts

```diff
@@ -25,12 +29,19 @@ export interface Sampler {
   /**
    * Checks whether span needs to be created and tracked.
    *
-   * TODO: Consider to add required arguments https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sampling-api.md#shouldsample
    * @param [parentContext] Parent span context. Typically taken from the wire.
    *     Can be null.
    * @returns whether span should be sampled or not.
    */
-  shouldSample(parentContext?: SpanContext): boolean;
+  shouldSample(
+    parentContext: SpanContext | undefined,
+    traceId: string,
+    spanId: string,
```

Comment:
Just so you know, `spanId` is going to be removed from the input to the sampler.

Reply:
Do you have any spec PR link to this? Do we expecting remove `spanId` in this PR?

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/621 just merged :)

The PR is named "spec compliant" so I would assume so ;)